src/WrapperRunner/SuiteLoader.php

```diff
@@ -73,8 +77,21 @@ public function __construct(
         $this->testCount = count($testSuite);
 
         $files = [];
-        $this->loadFiles($testSuite, $files);
+        $tests = [];
+        foreach ($this->loadFiles($testSuite) as $file => $test) {
+            $files[$file] = null;
+
+            if ($test instanceof TestCase) {
+                $name = str_replace(' with data set ', '', $test->nameWithDataSet());
```

Comment:
Unfortunately this doesn't work for both numeric and string data names.
`PHPUnit\Runner\Filter\Factory` needs:

1. `testName#1` for numeric names
2. `testName with data set "foo"` for string names

Reply:
looks like we could just use `$name = $test->nameWithDataSet());`

Reply:
@Guite no: `nameWithDataSet` outputs `testDataProvider1 with data set #0` for numeric names, and that doesn't work with `--filter`